geoopt/manifolds/siegel/vvd_metrics.py

```diff
@@ -0,0 +1,181 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+import torch
+
+
+class SiegelMetricType(Enum):
+    """Supported metric types for Siegel Spaces"""
+    RIEMANNIAN = "riem"
+    FINSLER_ONE = "fone"
+    FINSLER_INFINITY = "finf"
+    FINSLER_MINIMUM = "fmin"
+    WEIGHTED_SUM = "wsum"
+
+    @staticmethod
+    def from_str(label):
+        types = {t.value: t for t in list(SiegelMetricType)}
+        return types[label]
+
+
+class SiegelMetric(ABC):
+    """
+    Based on the vector-valued distance computed on Siegel spaces, different metric functions
+    can be taken, which give raise to different distances that can be computed in the space.
+
+    The vector-valued distance is given by :math:`v_i = log((1 + e_i) / (1 - e_i)), i = 1, ..., n`,
+    with :math:`e_i` the eigenvalues of the crossratio matrix sorted in ascending order
+    (:math:`e_1 < e_2 < ... < e_n`), and :math:`n = rank`.
+
+    Parameters
+    ----------
+    rank : int
+         Rank of the spaces. Only mandatory for Finsler distance of minimum entropy or weighted sum.
+    """
+
+    def __init__(self, rank: int = None):
+        self.rank = rank
+
+    @abstractmethod
+    def compute_metric(self, v: torch.Tensor, keepdim=False) -> torch.Tensor:
+        raise NotImplementedError
+
+    @classmethod
+    def get(cls, type_str: str, rank: int):
+        metrics_map = {
```

Comment:
This can be a static class variable

Reply:
Since I cannot have the children classes declared in a static class variable in the parent class, I extracted the creation logic in a `SiegelMetricFactory`